rasa/core/channels/socketio.py

```diff
@@ -95,27 +95,29 @@ async def send_elements(
                 }
             }
 
-            await self._send_message(self.sid, message)
+            await self._send_message(recipient_id, message)
 
     async def send_custom_json(
         self, recipient_id: Text, json_message: Dict[Text, Any], **kwargs: Any
     ) -> None:
         """Sends custom json to the output"""
 
-        json_message.setdefault("room", self.sid)
+        json_message.setdefault("room", recipient_id)
 
         await self.sio.emit(self.bot_message_evt, **json_message)
 
     async def send_attachment(
         self, recipient_id: Text, attachment: Dict[Text, Any], **kwargs: Any
     ) -> None:
         """Sends an attachment to the user."""
-        await self._send_message(self.sid, {"attachment": attachment})
+        await self._send_message(recipient_id, {"attachment": attachment})
 
 
 class SocketIOInput(InputChannel):
     """A socket.io input channel."""
 
+    sio = None
```

Comment:
Mhm, I'm a bit torn to be honest. Yes it works now as we are only using one Sanic worker (= 1 process). As soon as we are using multiple Sanic workers or multiple Rasa Open Source instances, this will break (meaning that the message is lost).